app/src/main/java/com/mrudultora/colorpickerapp/ColorPickerDialogsFragment.java

```diff
@@ -97,8 +98,8 @@ public void onClick(View view) {
     }
 
     public void selectColorDialog() {
-        colorPickerDialog = new ColorPickerDialog(getActivity());
-        colorPickerDialog.setColors()
+        colorPickerDialog = new ColorPickerBuilder(getActivity())
```

Comment:
Rather than creating a separate `ColorPickerBuilder` class, I think we must go for a static inner class `Builder`. This is because of 2 reasons. Firstly, using `ColorPickerBuilder` directly, one may get confused about its usage. For example,
```java
new ColorPickerDialog.Builder(context).show();
```
Secondly, that will keep all the code in same class, rather than moving it to separate classes.

Reply:
I agree, but the goal is to share configuration between dialog and bottom sheet and avoid duplication

Reply:
we may use 
```java
ColorPickerDialog.builder(context).show();
```

where builder is static method that redirect to the builder

Reply:
Yes, so that it can be used by both in the way you stated.
```java
new ColorPickerDialog.builder(context).show();
new ColorPickerBottomSheetDialog.builder(context).show();
```

Reply:
yes but without `new` since it will be a method

Reply:
I like this builder pattern but if it's a method, I think we must go for code duplication and implement a static `Builder` inner class in both. These are my personal views. Because if it would be a method then a dev may not get freehand. Your views ?

Reply:
it's a method in the sense it will just forward to the shared builder class

```java

class ColorPickerBuilder {}

class ColorPickerDialog {
    static ColorPickerBuilder builder(Context context) {
        return new ColorPickerBuilder(context, ColorPickerBuilder.Type.Dialog);
    }
}

class ColorPickerBottomSheetDialog {
    static ColorPickerBuilder builder(Context context) {
        return new ColorPickerBuilder(context, ColorPickerBuilder.Type.Sheet);
    }
}

```